src/iter.rs

```diff
@@ -199,6 +201,26 @@ impl<W: tokio::io::AsyncSeek + Unpin> tokio::io::AsyncSeek for ProgressBarIter<W
         Pin::new(&mut self.it).poll_complete(cx)
     }
 }
+#[cfg(feature = "tokio")]
+impl<W: tokio::io::AsyncBufRead + Unpin + tokio::io::AsyncRead> tokio::io::AsyncBufRead
+    for ProgressBarIter<W>
+{
+    fn poll_fill_buf(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<io::Result<&[u8]>> {
+        let pb = self.progress.clone();
+        Pin::new(self.get_mut().it.borrow_mut())
+            .poll_fill_buf(cx)
+            .map(|x| {
+                x.map(|buf| {
+                    pb.inc(buf.len() as u64);
+                    buf
+                })
+            })
```

Comment:
We can spell it like this:

```suggestion
        let this = self.get_mut();
        let result = Pin::new(&mut this.it).poll_fill_buf(cx);
        if let Poll::Ready(Ok(buf)) = &result {
            this.progress.inc(buf.len() as u64);
        }
        result
```

This is a fair bit more compact, easier to follow (IMO) and avoids the `ProgressBar` clone. What do you think?

Reply:
I agree, it was pretty workaroundish in my version, just to satisfy the borrow checker